### ISSUE TITLE
Promisified create/save dont fail when overwritting same key

### ISSUE THREAD

**Issue by jacintoArias** [automategreen/dynamoose#205]
Minimal example:

```
const testSchema = new dynamoose.Schema({
  id: {
    type: String,
    required: true,
    hashKey: true,
  },
  name: {
    type: String,
    required: true,
  },
});

const TestModel = dynamoose.model('test', testSchema);

const item1 = {
  key: "0000",
  name: "Item 1"
};

Device.create(item1, { overwrite: false })
   .then((item) => console.log(item))
   .catch((err) => console.log(err));

Device.create(item1, { overwrite: false })
   .then((item) => console.log(item))
   .catch((err) => console.log(err));
```

Behaviour:

```
NewModel { id: '0000', name: 'Item 1' }
undefined
```

Expected behaviour:

```
NewModel { id: '0000', name: 'Item 1' }
<Some Error>
```

I do not know if this is the expected behaviour, but I would prefer to tackle a failed overwrite by using catch as it is easy to return a failed response in my API.

This might be related to #186, but I can confirm that I'm getting other exceptions such as 'Required value'.

**Comment by brandongoode** [comment]
@jacintoArias I think this may be corrected in #200.   I just merged it into the master branch.  Could you try updating dynamoose to the current master and let me know if you still have the issue?

**Comment by jacintoArias** [comment]
The updated package now trows the expected exception and returns the following error:

```
NewModel { id: '0000', name: 'Item 1' }
The conditional request failed
```

thanks!
